ctms/acoustic_service.py

```diff
@@ -0,0 +1,273 @@
+        "fxa_lang": 1,
+        "fxa_login_date": 1,
+        "fxa_first_service": 1,
+        "fxa_created_date": 1,
+        "fxa_primary_email": 1,
+        "double_opt_in": 1,
+        "has_opted_out_of_email": 1,
+        "mailing_country": 1,
+        "first_name": 1,
+        "last_name": 1,
+        "sumo_contributor": 1,
+        "sumo_user": 1,
+        "sumo_username": 1,
+        "basket_token": 1,
+        "unsubscribe_reason": 1,
+    }
+
+    MAIN_TABLE_SUBSCR_FLAGS = {
+        # maps the Basket/CTMS newsletter name to the Acoustic name
+        "common-voice": "sub_common_voice",
+        "mozilla-fellowship-awardee-alumni": "sub_mozilla_fellowship_awardee_alumni",
+        "firefox-accounts-journey": "sub_firefox_accounts_journey",
+        "firefox-news": "sub_firefox_news",
+        "mozilla-and-you": "sub_firefox_news",
+        "hubs": "sub_hubs",
+        "internet-health-report": "sub_internet_health_report",
+        "knowledge-is-power": "sub_knowledge_is_power",
+        "miti": "sub_miti",
+        "mixed-reality": "sub_mixed_reality",
+        "mozilla-festival": "sub_mozilla_festival",
+        "mozilla-foundation": "sub_mozilla_foundation",
+        "mozilla-technology": "sub_mozilla_technology",
+        "mozillians-nda": "sub_mozillians_nda",
+        "take-action-for-the-internet": "sub_take_action_for_the_internet",
+        "test-pilot": "sub_test_pilot",
+        "about-mozilla": "sub_about_mozilla",
+        "app-dev": "sub_apps_and_hacks",
+    }
+
+
+class Acoustic(Silverpop):
+    """
+    :param client_id
+    :param client_secret
+    :param refresh_token
+    :param server_number
+    """
+
+    @staticmethod
+    def _process_response(resp):
+        logger.debug("Response: %s", resp.text)
+        response = etree.fromstring(resp.text.encode("utf-8"))
+        # success = response.find(".//SUCCESS")
+
+        # how main table failures are reported:
+        fault = response.find(".//Fault/FaultString")
+        if fault is not None:
+            print(fault.text)
+            raise SilverpopResponseException(fault.text)
+
+        # how RT failures are reported:    (hooray for consistency.)
+        failures = response.findall(".//FAILURES/FAILURE")
+        if failures is not None:
+            for fail in failures:
+                raise SilverpopResponseException(etree.tostring(fail))
+
+        return response
+
+    def _call(self, xml):
+        logger.debug("Request: %s", xml)
+        # Following basket's suggested pattern,
+        # adding force_bytes to deal with some Acoustic rejections
+        response = self.session.post(self.api_endpoint, data={"xml": force_bytes(xml)})
+        return self._process_response(response)
+
+
+class CTMSToAcousticService:
+    def __init__(
+        self,
+        client_id=os.environ["ACOUSTIC_CLIENT_ID"],
+        client_secret=os.environ["ACOUSTIC_CLIENT_SECRET"],
+        refresh_token=os.environ["ACOUSTIC_REFRESH_TOKEN"],
```

Comment:
expected secrets embedded in env vars, currently. 

What is the approach for secrets on background machinery?